Expose a notification switch to HA and pulse the top LED's when on.

andrew-codechimp opened the pull request:
As per title, a switch in HA will cause the top LED's to pulse when on, returning to either wake word or off if the switch is off.

Demo [here](https://tube.tchncs.de/w/bAFAckpGFG8VCLephv8wXs)

alex-savin commented:
@andrew-codechimp Could you please share your automation code
Thank you in advance

andrew-codechimp commented:
@alex-savin Here you go, I also use the little unique todo script to stop duplicate entries getting added. 

Play notifications gist
https://gist.github.com/andrew-codechimp/3fe6863ea7addf939ec80f786cda9093

Add unique todo gist
https://gist.github.com/andrew-codechimp/2d348f9b35ddbea526b88102c972539d

tetele commented:
First off, apologies for the delay, I've been on vacation last week.

Second, although I absolutely love your use case, I feel as though this is something very particular (which also requires some automation) and not related to the overall voice experience.

Instead of exposing a notification switch, I think it's a better approach to expose a pair of services (e.g. `onju_voice.notification_on` and `onju_voice.notification_clear`) which internally make that that light blink. I realize it's more complicated to use, but a switch has its meaning and purpose and I don't think it's a good choice here.

andrew-codechimp commented:
No problems for the delay, hope you had a good break.

Interesting idea to use services, switches add instant gratification in the device UI but services would be better for this as it's only ever triggered by automations.

Happy to re-work this to use services as it'll be a learning experience within ESPHome but if I make the change will you consider incorporating it?  I'm sure me and others are going to diverge from the default code in some ways so want to  understand the boundaries of what you'll incorporate in the default implementation.

tetele commented:
I would have surely incorporated such a feature when/if Assist would support bi-lateral conversation initiation. At the moment it's maybe a less common use case, although I could see it incorporated.

I get the gratification of using a switch, but again, I don't think the user needs a switch he could toggle on/off from the UI. Instead, I think the proper way is to call a service ([like these](https://github.com/tetele/onju-voice-satellite/blob/427110b72a44cf84ecbf9e3e0a61ad9e148b4625/esphome/onju-voice.yaml#L71-L77)) informing the device that it needs to show a notification beacon (be that a light or a beep or whatever else), then calling another service (either internally and/or from HA) to clear the notification flag. If you refactor it to use such services, I'd be happy to include it.

Please consider a modular approach to encapsulating what exactly should happen when the service gets called, as that would make future modifications easier. I'm thinking of `script`s here.

andrew-codechimp commented:
Added turn_on_notification and clear_notification services.
Added esphome common status binary sensor for diagnostics, that way HA automations can check devices re-connecting and set their notification or other states without storing in flash/having sync issues.

tetele commented on esphome/onju-voice-microwakeword.yaml:
```diff
@@ -241,6 +250,9 @@ esp32_touch:
   denoise_cap_level: L0
 
 binary_sensor:
+  - platform: status
```
And what value does that sensor have when the device is not connected? 'Cause I'm guessing it's `unavailable` instead of `off`.

In which case, why not check (from HA) if any exposed entity is `unavailable` instead of adding a new sensor? If you're checking from the ESPHome side, I don't understand where and there are other built-in sensors you can use.

andrew-codechimp replied:
Good point, the status sensor doesn't add anything to checking an existing sensor, saves bloating with extra sensors.
Removed and I've reworked my automations to look at switch.user_wake_word state, it's just as fast to change state as the status when you pull/apply the power.

tetele commented:
Thank you very much @andrew-codechimp <3

andrew-codechimp commented:
Thanks for adding this feature, it was interesting getting your feedback and I've learnt things along the way.
